runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -238,10 +252,66 @@ public void addDestination(String logicalGroup, List<String> remoteDestinations)
     }
 
     /**
-     * Remove a destination from the LogReplicationModelMetadataTable for a specific group.
+     * Sets the destinations for a given logical group to the provided list.
+     * <p>
+     * A new logical group will be created if one does not already exist.
+     * Setting destinations for a logical group will overwrite existing destinations.
+     * Can be used to clear existing destinations by passing in an empty list.
+     * Clearing existing destinations will remove the logical group.
+     *
+     * @param logicalGroup The group of tables to be replicated, represented as a string.
+     * @param remoteDestinations The clusters which the specified group will be replicated to.
+     * @throws IllegalArgumentException If logicalGroup is null or empty, if remoteDestinations
+     * is null, or if remoteDestinations contains null or empty elements.
+     */
+    public void setDestination(String logicalGroup, List<String> remoteDestinations) {
+        Preconditions.checkArgument(isValid(logicalGroup),
+                String.format("[%s] logicalGroup is null or empty.", clientName));
+        Preconditions.checkArgument(remoteDestinations != null,
+                String.format("[%s] remoteDestinations is null.", clientName));
+        Preconditions.checkArgument(hasNoNullOrEmptyElements(remoteDestinations),
+                String.format("[%s] remoteDestinations contains null or empty elements.", clientName));
+        List<String> finalRemoteDestinations = new ArrayList<>(deduplicate(remoteDestinations));
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = buildClientDestinationInfoKey(logicalGroup);
+                    DestinationInfoVal clientDestinations = DestinationInfoVal.newBuilder()
+                            .addAllDestinationIds(finalRemoteDestinations)
+                            .build();
+
+                    if (finalRemoteDestinations.isEmpty()) {
+                        log.info(String.format("[%s] Empty logical group %s, will be removed.", clientName, logicalGroup));
+                        txn.delete(sourceMetadataTable, clientInfoKey);
+                    } else {
+                        if (txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload() != null) {
+                            log.info(String.format("[%s] Existing destinations are being overwritten.", clientName));
```

Comment:
If you're anyways reading the entry, I think for debugging purposes it might be good to log the set of destinations to be overwritten and by what.. this could be helpful during debugging.